tests/integration/attendanceSheets.test.js

```diff
@@ -737,6 +737,89 @@ describe('ATTENDANCE SHEETS ROUTES - GET /attendancesheets', () => {
   });
 });
 
+describe('ATTENDANCE SHEETS ROUTES - PUT /attendancesheets/{_id}', () => {
+  let authToken;
+  beforeEach(populateDB);
+
+  describe('TRAINER', () => {
+    beforeEach(async () => {
+      authToken = await getToken('trainer');
+    });
+
+    it('should update an attendance sheet for a single course', async () => {
+      const attendanceSheetId = attendanceSheetList[5]._id;
+      const payload = { slots: [slotsList[4]._id] };
+
+      const response = await app.inject({
+        method: 'PUT',
+        url: `/attendancesheets/${attendanceSheetId}`,
+        headers: { Cookie: `alenvi_token=${authToken}` },
+        payload,
+      });
+
+      expect(response.statusCode).toBe(200);
+
+      const attendanceSheetUpdated = await AttendanceSheet.countDocuments({ _id: attendanceSheetId, ...payload });
+      expect(attendanceSheetUpdated).toEqual(1);
+    });
+
+    it('should return 404 if attendance sheet doesn\'t exist', async () => {
+      const payload = { slots: [slotsList[4]._id] };
+
+      const response = await app.inject({
+        method: 'PUT',
+        url: `/attendancesheets/${new ObjectId()}`,
+        headers: { Cookie: `alenvi_token=${authToken}` },
+        payload,
+      });
+
+      expect(response.statusCode).toBe(404);
+    });
+
+    it('should return 403 if subProgram is not include in single course', async () => {
```

Comment:
j'ajouterai "subprograms" a la fin : `should return 403 if subProgram is not include in single course subprograms`

Reply:
je pense qu'on peut simplement mettre "should return 403 if is course is not single". Le sous-programme c'est la manière qu'on aujourd'hui de vérifier ça mais à terme ça sera juste le type de la formation qu'on vérifiera

Reply:
✅